articles/article-112-compiler.md

```diff
@@ -0,0 +1,32 @@
+<!-- BEGIN TITLE -->
+# We have Machines within Machines
+<!-- END TITLE -->
+
+<!-- BEGIN BODY -->
+In 1954, computers just computed numbers. Grace Hopper, a famous
+female programmer around then, had a hard time to convince her
+colleagues that these machines can do more.
+
+The picture is hers and shows how a computer program can translate
+one computer program into another one. She even coined the term that is still in use
+today: a **compiler**.
+
+Nowadays, we have many more of these translation and transformation
+machines, with names as funny as the picture: _cross-compilers_,
+_transpilers_, and _compiler-compilers_ (cocos). Their mission is to
+translate from one computer language to another, for example
```

Comment:
I realized I made a correction that has the exact same line as what you have hear, in which case you can just drop that part and say:
These compilers can translate many things, for example...